Documentation

nschimmoller opened the pull request.

nschimmoller commented:
@zware okay breaking #183 into a smaller PRs and rebasing to make commit list less crazy. Mind taking a look?

nschimmoller commented:
@zware  happy new year! wanted to see if you'd have some time soon to review some of the open PRs of mine. Want to avoid making PRs without your or Christian's review in the future.

zware commented:
So sorry I haven't made it back here yet, unclaimed time at my desk has not existed lately.  Will review as soon as I have a chance.

zware reviewed (changes requested):
I have not had time to fully review everything here, but I have some blocking comments and a couple of fixes in the form of #198 and #199 that will impact this as well.

> Reviewed 3 of 11 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/pycountry/pycountry/195#-:-NnbFkxA9k1ASanfvUj8:bgwymj8)* status: 3 of 11 files reviewed, 3 unresolved discussions (waiting on @nschimmoller)

___
*[`CODEOFCONDUCT.md` line 53 at r1](https://reviewable.io/reviews/pycountry/pycountry/195#-NnUxL-W0ywDQGuehCVn:-NnUxL-W0ywDQGuehCVo:berb728) ([raw file](https://github.com/pycountry/pycountry/blob/83d10e769fd6010d25f02d7883dd6317926a0544/CODEOFCONDUCT.md#L53)):*
> ```Markdown
> ## Attribution
> 
> This Code of Conduct is adapted from the [Contributor Covenant](https://www.contributor-covenant.org/), version 2.0, available at [https://www.contributor-covenant.org/version/2/0/code_of_conduct.html](https://www.contributor-covenant.org/version/2/0/code_of_conduct.html).
> ```

I don't actually see much relationship between this file and Contributor Covenant v2.0.  It looks a little closer to v1.4, but it's still pretty far off from it.

Did this text actually originate from contributor-covenant.org?
___
*[`CONTRIBUTING.md` line 3 at r1](https://reviewable.io/reviews/pycountry/pycountry/195#-NnXDz20AskPpTNhL5EM:-NnXDz216cQXRshJV8TW:bdehr55) ([raw file](https://github.com/pycountry/pycountry/blob/83d10e769fd6010d25f02d7883dd6317926a0544/CONTRIBUTING.md#L3)):*
> ```Markdown
> # Contributing to pycountry
> 
> Welcome to the pycountry project! We appreciate your interest in contributing. Whether it's improving documentation, adding new features, fixing bugs, or helping with testing, your contributions are valuable to the community.
> ```

I'm really going to have to insist on shorter lines.  Some tools will try to make long lines more readable by choosing an arbitrary place to wrap, but it's never consistent and not all tools do.  I'm ok with either wrap at 80 or [semantic wrapping](https://sembr.org/), but I can't accept 300 column lines.
___
*[`CONTRIBUTING.md` line 28 at r1](https://reviewable.io/reviews/pycountry/pycountry/195#-NnVbCiA4GDpUzNTp-eb:-NnVbCiB8Bg8wUfA_ay2:bfgares) ([raw file](https://github.com/pycountry/pycountry/blob/83d10e769fd6010d25f02d7883dd6317926a0544/CONTRIBUTING.md#L28)):*
> ```Markdown
> 2. Clone your fork locally using ``git clone <your-fork-url>``.
> 3. Navigate to the cloned directory: ``cd pycountry``.
> 4. Install the project and its dependencies: ``pip install -e .`` (This installs the package in editable mode).
> ```

This step isn't necessary and, without recommending a venv beforehand, encourages the pollution of one's global environment, which is less than ideal.


<!-- Sent from Reviewable.io -->

nschimmoller commented:
## Updates based on review feedback:
- Changed CODEOFCONDUCT.md attribution from "adapted from" to "inspired by the Contributor Covenant"
- Wrapped long lines in CONTRIBUTING.md to ~80 columns using semantic line breaks
- Added virtual environment setup step in CONTRIBUTING.md before pip install -e .
- Updated SECURITY.md to use sustainable version policy ("Latest stable release" / "Previous major version") instead - of hardcoded version numbers
- Removed reference to non-existent CONTRIBUTORS.txt from PR template

Ready for rereview @zware
